types/types.go

```diff
@@ -355,3 +355,34 @@ func (jm JSONMap) GormValue(ctx context.Context, db *gorm.DB) clause.Expr {
 	data, _ := jm.MarshalJSON()
 	return gorm.Expr("?", string(data))
 }
+
+// GenericStructValue can be set as the Value() func for any json struct
+func GenericStructValue[T any](t T, defaultNull bool) (driver.Value, error) {
```

Comment:
Is this name okay or should I rename it to `GenericStructJSONValue` ?

Reply:
fine with both 👍